### DISCUSSION TITLE
Metrics on inflight-window/message queue

### DISCUSSION THREAD

**Discussion by ennc0d3** [emqx/emqx#13556] [General]
I am looking for a counter which can give the state of the inflight window and message queue so that I could setup some alerts. I can see them in the dashboard but could not see any corresponding metric in the document.

The rationale, is that if the inflight-window is full, the message queue builds up and so the message delivery goes at the rate of the infligh-windows's clearing. It would be good to see them early so we can setup alerts.

Thanks, Naga

**Comment by savonarola** [comment]
Hello! Inflight-window size is client-specific, so there is no cluster-wide metric for that.

Maybe [Slow Subscriptions](https://docs.emqx.com/en/emqx/latest/observability/slow-subscribers-statistics.html) feature may help?

**Reply by ennc0d3** [reply]
Though Slow Subscription help to identify the issue, its more of reactive approach. Thanks
